### PR TITLE
Change 'trimRight()' to 'trim()' for better browser support.

### PR DESCRIPTION
**Pull request by yipingliao**
Also, call 'formatTitleFromUrl' for recommendation attributions only if there is a url.
https://app.asana.com/0/15589946826415/15785999026810

I found a bug while testing recommendations on IE10 where the browser does not recognize the `trimRight()` method on JavaScript strings. This is also a problem for keeps, but the code path that contains `trimRight()` is accessed more frequently for recommendations. Switching to `trim()`, which is supported by IE9 and up.

@2is10 cc @stkem

### CODE REVIEW

File: kifi-backend/modules/shoebox/angular/src/keeps/keepActionService.js [hunk 1]
```diff
@@ -33,7 +33,7 @@ angular.module('kifi')
           };
         })
       };
-      $log.log('keepService.keep()', data);
+      $log.log('keepActionService.keep()', data);
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/keeps/keepActionService.js:36]: This is just for clarification in logging.

File: kifi-backend/modules/shoebox/angular/src/recos/recoDecoratorService.js [hunk 2]
```diff
@@ -29,7 +29,7 @@ angular.module('kifi')
         fileNameIdx += fileNameMatch.index;
         fileName = fileNameMatch[0];
       }
-      fileName = fileName.replace(fileNameToSpaceRe, ' ').trimRight().trimLeft();
+      fileName = fileName.replace(fileNameToSpaceRe, ' ').trim();
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/recos/recoDecoratorService.js:32]: On the branch where I am doing refactoring, all 3 instances of `trim()` is DRYed up into one.

File: kifi-backend/modules/shoebox/angular/src/recos/recoDecoratorService.js [hunk 3]
```diff
@@ -126,7 +126,7 @@ angular.module('kifi')
       };
 
       this.recoData.reasons.forEach(function (reason) {
-        if (!reason.name) {
+        if (!reason.name && reason.url) {
```
Comment (yipingliao) [inline kifi-backend/modules/shoebox/angular/src/recos/recoDecoratorService.js:129]: Only convert url to title if there is a url to begin with.

**Comment by 2is10** [comment]
Great!

**Pull request merged**